core/lib/dal/src/blocks_dal.rs

```diff
@@ -2159,6 +2172,40 @@ impl BlocksDal<'_, '_> {
         .await?;
         Ok(())
     }
+
+    pub async fn get_tree_writes(
+        &mut self,
+        l1_batch_number: L1BatchNumber,
+    ) -> DalResult<Option<Vec<TreeWrite>>> {
+        let tree_writes: Option<Vec<u8>> = sqlx::query!(
+            r#"
+            SELECT
+                tree_writes
+            FROM
+                l1_batches
+            WHERE
+                number = $1
+            "#,
+            i64::from(l1_batch_number.0),
+        )
+        .instrument("get_tree_writes")
+        .with_arg("l1_batch_number", &l1_batch_number)
+        .fetch_optional(self.storage)
+        .await?
+        .and_then(|row| row.tree_writes);
+
+        let tree_writes = tree_writes
+            .map(|data| {
+                bincode::deserialize(&data).map_err(|err| {
+                    Instrumented::new("get_tree_writes")
+                        .with_arg("l1_batch_number", &l1_batch_number)
+                        .constraint_error(err.into())
+                })
+            })
+            .transpose()?;
```

Comment:
This can be done more fluently and idiomatically by using `.try_map()` like [e.g. here](https://github.com/matter-labs/zksync-era/blob/33000475b47831fc3791dac338aae4d0e7db25b0/core/lib/dal/src/blocks_dal.rs#L2014).